### PR TITLE
[IS-3599] Fixed crash issue when editing a message

### PR DESCRIPTION
**Pull request by aliabbasmalik8**
### Fixed Issues

$ https://github.com/Expensify/App/issues/3599

### QA Steps

- Find your message in the conversation
- Click on Edit
- Write a longer message

 ### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

https://user-images.githubusercontent.com/31027036/126577779-8132b94f-9fc0-4283-a347-350b0bdb1b34.mov

#### Desktop

https://user-images.githubusercontent.com/31027036/126577913-0a6ce524-7e73-4844-b18e-90a716f2e3dc.mov

#### iOS

https://user-images.githubusercontent.com/31027036/126577497-3334c964-2bfd-492d-aeb4-60963aa3e751.mov

#### Android

https://user-images.githubusercontent.com/31027036/126577622-5aa32ae6-baec-48a4-aaa7-dac56ac0ea79.mp4

### CODE REVIEW

File: src/pages/home/report/ReportActionItemMessageEdit.js [hunk 1]
```diff
@@ -68,9 +68,8 @@ class ReportActionItemMessageEdit extends React.Component {
      */
     updateDraft(newDraft) {
         this.textInput.setNativeProps({text: newDraft});
-        const trimmedNewDraft = newDraft.trim();
```
Comment (stitesExpensify) [inline src/pages/home/report/ReportActionItemMessageEdit.js:71]: Why are we just getting rid of the trim? I assume it was there for a reason right?

Comment (aliabbasmalik8) [reply src/pages/home/report/ReportActionItemMessageEdit.js:71]: we used selection in TextInputFocusable props so when we update the message then selection(cursor position) also update and if we use space at the end of the message then trim will remove that space so selection(cursor position) goes beyond the message length so it'll create issue

Comment (stitesExpensify) [reply src/pages/home/report/ReportActionItemMessageEdit.js:71]: So would a better solution then be to change the `cursor position`? Something like this maybe?

```
updatedCursorPosition = cursorPosition > message.length ? message.length : cursorPosition;
```

Comment (stitesExpensify) [reply src/pages/home/report/ReportActionItemMessageEdit.js:71]: I think it makes sense to trim drafts and get rid of useless whitespace in messages

Comment (aliabbasmalik8) [reply src/pages/home/report/ReportActionItemMessageEdit.js:71]: @stitesExpensify any update about the above-mentioned solution?

**Review by stitesExpensify — APPROVED** [review]

**Pull request merged**